sky/core.py

```diff
@@ -552,14 +555,12 @@ def cancel(
         sky.exceptions.CloudUserIdentityError: if we fail to get the current
           user identity.
     """
-    if not job_ids and not all:
-        raise ValueError(
-            'sky cancel requires either a job id '
-            f'(see `sky queue {cluster_name} -s`) or the --all flag.')
-
     backend_utils.check_cluster_name_not_reserved(
         cluster_name, operation_str='Cancelling jobs')
 
+    if all and job_ids:
```

Comment:
For job_ids, do we mean `is not None` or is not empty list?

Reply:
Yes. Otherwise, the cli will have an issue.